generator/src/main/java/com/scottlogic/deg/generator/outputs/datasetwriters/DataSetWriter.java

```diff
@@ -7,13 +7,13 @@
 import java.io.IOException;
 import java.nio.file.Path;
 
-public interface DataSetWriter<TWriter extends Closeable> {
+public interface DataSetWriter<TWriter extends Closeable, C> {
```

Comment:
I know single-letter generic types are common in Java, but they make it really hard to figure out the intent of a type.